source/agora/common/crypto/Schnorr.d

```diff
@@ -209,6 +212,25 @@ public bool verify (T) (const ref Point X, const ref Signature s, auto ref T dat
     return S == RcX;
 }
 
+/// Publickey(Ed25519) to Point(Curve25519)
+public static Point pkToCurvePoint (PublicKey pk) nothrow @nogc
+{
+    ubyte[crypto_sign_ed25519_PUBLICKEYBYTES] ed25519_pk;
+    if (crypto_sign_ed25519_pk_to_curve25519(ed25519_pk[].ptr, pk[].ptr) != 0)
+        assert(0);
+    return Point(ed25519_pk);
+}
+
+///
+unittest
+{
+    KeyPair kp = KeyPair.fromSeed(
+        Seed.fromString(
+            "SCT4KKJNYLTQO4TVDPVJQZEONTVVW66YLRWAINWI3FZDY7U4JS4JJEI4"));
+    Point point = pkToCurvePoint(kp.address);
```

Comment:
Can you add the `Scalar` corresponding to this `Point` ?

Reply:
It's been solved with the help of Jay.